entity-service-impl/src/main/java/org/hypertrace/entity/query/service/EntityQueryServiceImpl.java

```diff
@@ -827,8 +827,10 @@ private void doBulkUpdate(
 
       if (keys.isEmpty()) {
         // Nothing to update
-        LOG.info("No entity found with filter {} for updating", update.getFilter());
-        continue;
+        throw new IllegalArgumentException(
```

Comment:
Hmm - I know I reported this, but I'm less sure we actually want to treat this as an error now that I see this. 

I think there's a valid use case to want to update based on some criteria, and not have it error out if it doesn't hit anything. That said, if we go that way, I'd expect it to return the updated count if it doesn't already (so the caller can decide if zero matches is an error to it or not) - and I'm guessing by the void return on this method that it doesn't do that.

So I think returning the update count would be better than an error, but if that's not feasible I guess we can stick with the error for now.

Reply:
Sure, Aaron. That makes sense. Updated to return the updated ids.